Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/publish.py

```diff
@@ -235,10 +240,23 @@ def main(cls, args, repository, hooks=None, **kwargs):
         push_env = os.environ.copy()
         push_env['VERBOSITY'] = str(args.verbose)
         push_env['PUSH_HOOK_MODE'] = 'publish'
+        remote_arg = args.remote
+
+        if args.user:
+            remote = repository.remote(remote_arg)
+            if not remote or not remote.checkout_url(http=True) or not getattr(remote, 'domain', None):
+                sys.stderr.write("Cannot convert '{}' to an ephemeral HTTP url\n".format(remote_arg))
+                return 1
+            remote_arg = remote.checkout_url(http=True)
```

Comment:
It doesn’t look like `checkout_url` can return a falsey value?

```suggestion
            if not remote or not getattr(remote, 'domain', None):
                sys.stderr.write("Cannot convert '{}' to an ephemeral HTTP url\n".format(remote_arg))
                return 1
            remote_arg = remote.checkout_url(http=True)
```

(this also avoids a redundant call)